Mamma-mia! Adds-a the Italian-a language!

Ordonis opened the pull request:
:cl: Ordo
rscadd: Mamma-mia! The chef speaks-a so different now!
/:cl:

Adds italian. WIP. Will make chef brain in separate commit.

Edit: Now being tested. Initial Italianization ETA: as soon as I finish my SPAGHETT.

SPAGHETT finished. Ready for Pizza deployment.

Ordonis committed:
Mamma-mia!

That's a one-a spicy meatball! The first of many spicy meatballs!

ExcessiveUseOfCobblestone commented:
:bike: :house: add me to the ss

MMMiracles commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"operative","greek")
+		message = replacetext(message,"op","greek")
+		message = replacetext(message,"ops","greeks")
+		message = replacetext(message,"nuke","the spiciest-a meatball")
```
the greeks armed the spiciest-a meatball

Ordonis replied:
I knew-a the gifts they were-a bringing were too good to be-a true!

Ordonis commented:
But cobby, you can't ss with me! My fiance would kill me! Also if you're referring to an actual git thing, I have no idea-a what you're-a talking about.

arsserpentarium commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"spicy","a-spicy")
+		message = replacetext(message,"I'm","I'm-a")
+		message = replacetext(message,"friend","enemy")
+		message = replacetext(message,"enemy","friend")
```
why?

Ordonis replied:
Two world wars

arsserpentarium replied:
Not very funny, but could be annoying in game.

JJRcop replied:
This is going to result in enemy being replaced by friend, and friend staying in tact, since you directly replace all friend with enemy, then after that enemy with friend, which includes all enemy and doesn't distinguish with enemy that used to be friend.

MMMiracles replied:
you should be able to just put an italian spin both words to prevent it

Ordonis replied:
I'll add the -a then

JJRcop replied:
ene-mia

Grazyn replied:
Actually, the triple alliance was defensive and the Austrians attacked first

ShizCalev replied:
paesano

Iamgoofball replied:
@grazyn go home nazi

ExcessiveUseOfCobblestone commented:
ss = **s**creen**s**hot

arsserpentarium reviewed (commented):
I think, it's rather stupid idea and implementation.sorry.

ExcessiveUseOfCobblestone commented:
arsse stop being a bigot against italians

Kromgar commented:
BORK BORK BORK BORK BORK BORK

Armhulen commented:
text_lose_indication = "<span class='notice'>You no longer feel the urge to switch sides midway through a war. Your love of pasta passes.</span>"

Has a pr ever come this close to perfection?

arsserpentarium commented:
I am not against italians.I just dislike idea.
Maybe you'd better code special language for african skinned chars in "YO NIGGA WASSUP"-style?

Ordonis commented:
Well, given my propensity to fuck up basic checks by having an out of date testing server, I want to focus on doing one thing at a time; Bixnoodian would have to come after getting Italian properly coded and the chef's brain properly mutated into a warm bowl of pasta.

Bombpants commented:
Can we get enough languages to turn nation rounds into NATION rounds?

ike709 commented:
@Bombpants I was a strong supporter of your idea until you gave yourself a thumbs up.

Hatfish commented:
>Italian
>NOT SWEDISH

repoban immediately

PKPenguin321 commented:
I made literally this exact same thing before but with spanish for janitors and got shut down

PKPenguin321 commented:
Oh wait this is a speak quirk not a full language hm

Repukan commented:
Somebody touch-a my spaghett

SpaceManiac commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"whose","whose-a")
+
+		if(prob(10))
+			message += " Ravioli [pick(", ravioli",", ravioli, linguini")]!"
```
This will be spaced oddly, like `Ravioli , ravioli`, Also end it in a full stop?

Ordonis replied:
I'll fix the spacing and add the punctuation in the next commit

81Denton replied:
Nafri LMFAO

Jordie0608 commented:
https://www.youtube.com/watch?v=V5R75TodGXc

Bawhoppen commented:
Just add "-a" to the end of every word for posterity's sake

yorii commented:
Funny PR, I love it.
May I give some additional suggestions?

`
message = replacetext(message,"hello",pick("ciao","buongiorno"))
message = replacetext(message,"bye",pick("ciao","arrivederci"))
message = replacetext(message,"that's","that's-a")
message = replacetext(message,"enough","basta")
message = replacetext(message,"try","trattare")
message = replacetext(message,"deal","trattare")
`

XDTM commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+	if(message)
+		message = " [message] "
+		message = replacetext(message,"mom","mamma")
+		message = replacetext(message,"dad","pappa")
```
papà

XDTM commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"why","for-a what reason")
+		message = replacetext(message,"my","my-a")
+		message = replacetext(message,"it's","it's-a")
+		message = replacetext(message,"bad","molto-mal")
```
molto male

XDTM commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"greytide","nafri")
+		message = replacetext(message,"captain","capitano")
+		message = replacetext(message,"sec","polizia")
+		message = replacetext(message,"security","polizia")
```
add shitcurity or shitsec -> carabinieri

ShizCalev replied:
hos -> fascista

XDTM commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"captain","capitano")
+		message = replacetext(message,"sec","polizia")
+		message = replacetext(message,"security","polizia")
+		message = replacetext(message,"cheese","parmesano")
```
parmigiano

Grazyn replied:
Honestly, I'm Italian and I find the incorrect words much more amusing than the correct ones that most players probably wouldn't even know how to pronounce. It gives the feeling of the stereotypical broken sing-song Italian accent I usually hear in american fiction, which I assume is the purpose of this PR

XDTM replied:
True, that's probably better

XDTM reviewed (commented):
Pure italian review

ShizCalev commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"what","what-a")
+		message = replacetext(message,"and","and-a")
+		message = replacetext(message,"am","am-a")
+		message = replacetext(message,"assistant","sicilian")
```
goombah

Ordonis replied:
Mamma-mia! How-a did I not think of-a such a thing!

Grazyn commented:
thanks>grazie (I think everyone knows that)

XDTM commented:
Yes>Si

FlattestGuitar commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"cheese","parmesano")
+		message = replacetext(message,"meat","pepperoni")
+		message = replacetext(message,"who's","who's-a")
+		message = replacetext(message,"who","who-a")
```
`who's` will get turned into `who's-a`, then right after in the next line it will be turned into `who-a's-a`.

Make sure the order of the `replacetext()` calls is right and they don't replace each others' text.

yorii replied:
Maybe just skip "who-a" as it just sounds entirely dumb and not even italian.

Ordonis replied:
Aye, all rip that out once power is restored to my area. Reee snowstorms.

81Denton commented:
You may not like it, but this is what a peak PR looks like.

Birdtalon commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+	quality = MINOR_NEGATIVE
+	dna_block = NON_SCANNABLE
+	text_gain_indication = "<span class='notice'>You feel-a the spicy meatballs and-a pasta singing to your-a soul!</span>"
+	text_lose_indication = "<span class='notice'>You no longer feel the urge to switch sides midway through a war. Your love of pasta passes.</span>"
```
👌

Buggy123 commented:
5/7 flawless pr

>@Bombpants I was strongly in favor of your idea until you gave yourself a thumbs up.

What's so bad about that?

Dorsisdwarf commented:
I'm going to have to insist you make it a 50-50 chance of italian or swedish, because *come on*

Spaceinaba commented:
this is the best fuckin PR I've ever seen

81Denton commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"operative","greek")
+		message = replacetext(message,"op","greek")
+		message = replacetext(message,"ops","greeks")
+		message = replacetext(message,"nuke","the spiciest-a meatball")
```
Wouldn't "the nuke" filter to "the the spiciest-a meatball"?
Also my sides are in orbit, PR of the year

SpinnerMaster commented:
PR-a of the century-a

Grazyn commented:
Fuck you>vaffanculo

Bombpants commented:
What other jobs can get languages thrown onto them?

Ordonis commented:
I could see a Russian/Western Cowboy bartender, French mime, Dutch botanists (DUUDE WEED LMAO); I wouldn't want to add too many to roles which need quick and clear communication, though. Thankfully, the chef does not need quick and clear communication.

Grazyn commented:
@Bombpants crosspost from the forums

chef-italian
janitor-spanish
barman-cockney/scottish
scientist-asian
doctor/security-german
lawyer-yiddish
dark skinned-ebonics (assistant only, otherwise job-specific accent)

JJRcop commented:
When it comes to replacetext conflicts with itself, you can #define an intermediate magic string that starts with < like `#define IT_L_SPECIAL "<ITLang"` then you can for example, safely do
```DM
message = replacetext(message, "enemy", IT_L_SPECIAL)
message = replacetext(message, "friend", "enemy")
messsage = replacetext(message, IT_L_SPECIAL, "friend"
```

Iamgoofball commented:
@Grazyn thats racist as hell

Bawhoppen commented:
Only Janitor and Chef should have languages really. Stereotyping other jobs isn't as funny.

WJohn commented:
RIP could have been sweedish bork

CosmicScientist commented:
@PKPenguin321 that's because you didn't make the janitor polish and insert *", kurwa!"* before every exclamation.

Carbonhell commented:
I am italian and i find this amazing
If you need any help with phrases ask me on hippie

Carbonhell commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"sing","sing-a")
+		message = replacetext(message,"cook","cook-a")
+		message = replacetext(message,"want","want-a")
+		message = replacetext(message,"what's","what's-a")
```
Wont this become what-a's-a?

Carbonhell commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"assistant","sicilian")
+		message = replacetext(message,"greytide","nafri")
+		message = replacetext(message,"captain","capitano")
+		message = replacetext(message,"sec","polizia")
```
Wont this cause "security" to become "poliziaurity"

XDTM replied:
plenty of words use sec, might want to add spaces on the sides

Iatots replied:
how about "la curva" or "curvisti" for greytide instead of that german acronym?

Carbonhell commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
+		message = replacetext(message,"who's","who's-a")
+		message = replacetext(message,"who","who-a")
+		message = replacetext(message,"thing","thing-a")
+		message = replacetext(message,"whose","whose-a")
```
Instead of those manual replacement why not split the phrase in each word and have a like 30% chance to add -a to it?

Plapatin commented:
readd swedishness, bork bork bork

Potato-Masher commented:
So uh, you ever going to come back and work on this? Last time OP commented was over a week ago.

Ordonis commented:
Yes, I only finished repairing my computer yesterday

On Feb 2, 2018 8:58 AM, "Potato-Masher" <notifications@github.com> wrote:

> So uh, you ever going to come back and work on this? Last time OP
> commented was over a week ago.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/34916#issuecomment-362603435>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATHGG7WBcnu2wXX60i_aGMWhFMUYy6wVks5tQx3DgaJpZM4RsNqW>
> .
>

CosmicScientist commented:
For future's concern, you might need to make everyone who gains this language have a change in palette:
#35205

nicbn commented on code/datums/mutations/speech.dm:
```diff
@@ -230,3 +230,55 @@
 	owner.remove_language(/datum/language/beachbum)
+
+
+	/datum/mutation/human/italian
```
Indent

Kawaii-Big-Boss commented:
The chef is supposed to be swedish, I thought?

Bawhoppen commented:
The chef is supposed to be whatever your want their character to be, and this needs to just be an opt-in option when they actually do add it.

Ordonis committed:
I have-a returned from the exile and fixed-a the things people asked-a for!

Includes changing sec so that not every word with sec in it will end
with problems. Uses resident italian advice to make minor edits. The
chef is now mario-tier of not-italian.

Ordonis committed:
Adds-a the mutation code and-a calls it for the cook at roundstart

Mamma-mia whoever named-a the CLOWNMUT and-a made it a #define trolled-a
the hell out of me-a! It was like-a chasing an actual-a clown!

Ordonis commented:
Right, hopefully there isn't too much... SPAGHETTI CODE!

In all seriousness I have displayed a horrible ability to properly work testserbians, so a discerning eye would be appreciated.

Ordonis committed:
Actually bothers to upload the trait definition

Lesson gained: Save all, all the time.

Ordonis committed:
Case sensitive fun whee

Someday I'll learn to build first

Ordonis committed:
Merge branch 'master' into Italian-language

Ordonis committed:
IT COMPILES-A

Fucking finally.

Ordonis committed:
Small addition that was left out and attempting to merge resolve the odd DME change

I never actually edited the dme in any way, only dm files; I think I
just haven't updated the master recently, so the exe is in a different
place than the repo. Don't expect this to compile.

Ordonis committed:
Fixes .dme to match with master

If this doesn't work I'll start reverting to the last version that didn't throw a fit.

PKPenguin321 commented:
make an italian moustache mask and tie this to it
see: https://github.com/tgstation/tgstation/blob/master/code/modules/clothing/masks/boxing.dm#L22
chef can start with it equipped but take it off if he wants

Ordonis commented:
I want to get this done and properly working first; I'm still very much a novice when it comes to BYOND and its... peculiarities. I do think that is a good long-term goal, although I'm worried about feature creeping this too much and having it end up being a giant hornswoggle of pasta typing.

Ordonis committed:
Commit to get travis to recheck

.dmm should be fixed, so this needs rechecking. Also, adding wine
replacement.

Ordonis committed:
The pizza, she is a-ready to share-a with the people!

Fixes a lot, does as much sanitization as is possible so that words like
insect don't become inpoliziact or similar.  For some words, punctuation
may be displaced by the final space necessary to keep normal words from
running into eachother.  This part, at least, is done.

Ordonis commented:
Oranges pls no bully

coiax commented:
What the actual fuck

Iamgoofball commented:
Lmao @ people whining about preferences

This is not a democracy
